File: filter_action/filter_action.py
input = ["Zebra", "nice camel", "Horse"]

is_uppercase = True
make_uppercase = True

filter1 = lambda s: s[0] == "H"
filter2 = lambda s: s[0].isupper()

for f in [filter1, filter2]:
    input = [x for x in input if f(x)]

print(input)

File: filter_action/filter_action.rs

fn main() {
    let mut input: Vec<String> = vec!["Zebra", "nice camel", "horse"].iter().map(|&s| String::from(s)).collect();
    let mut filter = |s: &String| { true };
    // let mut filter = |s: &String| { s.chars().next().unwrap().is_uppercase() };
    if true { filter = |s: &String| { s.chars().next().unwrap().is_uppercase() }; }

    // let result: Vec<&String> = input.iter().filter(|&s| s.chars().next().unwrap().is_uppercase()).collect();
}
